pkg/store/proxy.go

```diff
@@ -388,18 +461,20 @@ func storeMatches(s Client, mint, maxt int64, matchers ...storepb.LabelMatcher)
 		return false, nil
 	}
 	for _, m := range matchers {
-		for _, l := range s.Labels() {
-			if l.Name != m.Name {
-				continue
-			}
+		for _, ls := range s.LabelSets() {
+			for _, l := range ls.Labels {
```

Comment:
hm, something is wrong here - or if right we need to comment it better.

Essentially this means that matching here treat set with `AND` operator and I think we need `OR` rather, right?

as:
`[a=1, b=1][a=1, b=2]` labelset should be *matched* for `a=1, b=2` matcher, but it will be fitered out by `if !m.Matches(l.Value) {`

Reply:
Yes you're right.

Reply:
Refactored this a bunch, I think this is much clearer now.